test/contracts/pass/cheatCodesFork.sol

```diff
@@ -0,0 +1,70 @@
+pragma experimental ABIEncoderV2;
+
+import "ds-test/test.sol";
+
+interface Hevm {
+    function warp(uint256) external;
+    function roll(uint256) external;
+    function load(address,bytes32) external returns (bytes32);
+    function store(address,bytes32,bytes32) external;
+    function sign(uint256,bytes32) external returns (uint8,bytes32,bytes32);
+    function addr(uint256) external returns (address);
+    function ffi(string[] calldata) external returns (bytes memory);
+    function prank(address) external;
+    function createFork(string calldata urlOrAlias) external returns (uint256);
+    function selectFork(uint256 forkId) external;
+}
+
+/// @dev This contract's state should depend on which fork we are on
+contract TestState {
+    uint256 public state;
+    function setState(uint256 _state) external {
+        state = _state;
+    }
+}
+/// @dev This contract's state should be persistent across forks, because it's the contract Echidna deploys
+contract ForkTest is DSTest {
+    Hevm hevm = Hevm(HEVM_ADDRESS);
+    address stateContract;
+    uint256 forkId1;
+    uint256 forkId2;
+    uint256 persistentState;
+
+    constructor() {
+        stateContract = address(new TestState());
+        forkId1 = hevm.createFork("foo"); // If the default fork's id is 0, then this would be 1
+        forkId2 = hevm.createFork("bar"); // and this would be 2
+        persistentState = 0;
+    }
+
+    function test_ForkedState() external {
```

Comment:
according to the foundry docs, the state of `msg.sender` should also be persisted betweeen forks. Could we also test that behaviour here (I guess by sending some eth to msg.sender and checking that it is persisted between forks?).